### PR TITLE
Fixes #2075 and #1949

### LINKED ISSUE #1949: Modal wrong action routing
https://github.com/the-control-group/voyager/blob/0b11c4c09d494ca72497ec0fc2e66dd21fa457d6/resources/views/menus/browse.blade.php#L111

Should be:
```
$('#delete_form')[0].action = '{{ route('voyager.'.$dataType->slug.'.index') }}/' + id;
```

Because if not, it's appending the id of the menu constantly.

Steps:
1) Click delete in any menu
2) Click cancel on the modal.
3) Click delete in the same menu or any other.

Result, sometime like that: http://voyager.dev/admin/menus/3/1/3

### PR DESCRIPTION
**Pull request by vblinden**
When deleting a menu, canceling it and opening another menu again. The action has two id's like this: .../2/1. This pull requests fixes that.

This will also fix #2075 and fix #1949

### CODE REVIEW

**Comment by fletch3555** [comment]
Can you make sure this applies to all types? This behavior likely exists in all Voyager BREAD views

**Comment by vblinden** [comment]
@fletch3555 Sure, I will check every delete modal and update the PR.

**Comment by fletch3555** [comment]
This is essentially the same as #2670.  One uses `__id`, and the other uses the index route and appends the id.  Can you make them function the same (I prefer using the correct route, not assuming the delete route is the same as appending the id to the index route).

**Comment by vblinden** [comment]
Fixed all the other instances I could find. Some have different routes and so we have to append the id to the end and not replace the `__id`.

File: resources/views/bread/browse.blade.php [hunk 1]
```diff
@@ -279,16 +279,7 @@
...
-
-            if (!deleteFormAction) { // Save form action initial value
-                deleteFormAction = form.action;
-            }
-
-            form.action = deleteFormAction.match(/\/[0-9]+$/)
-                ? deleteFormAction.replace(/([0-9]+$)/, $(this).data('id'))
-                : deleteFormAction + '/' + $(this).data('id');
-            console.log(form.action);
+            $('#delete_form')[0].action = '{{ route('voyager.'.$dataType->slug.'.index') }}' + '/' + $(this).data('id');
```
Comment (fletch3555) [inline resources/views/bread/browse.blade.php]: Can you make this one use the `__[x]` placeholder like the rest?

File: resources/views/menus/browse.blade.php [hunk 2]
```diff
@@ -106,9 +106,7 @@
         });
 
         $('td').on('click', '.delete', function (e) {
-            id = $(e.target).data('id');
-
-            $('#delete_form')[0].action += '/' + id;
+            $('#delete_form')[0].action = '{{ route('voyager.'.$dataType->slug.'.index') }}' + '/' + $(this).data('id');
```
Comment (fletch3555) [inline resources/views/menus/browse.blade.php]: This one too.

**Review by fletch3555 — CHANGES_REQUESTED** [review]

**Comment by fletch3555** [comment]
Sorry, just saw your comment about needing to append.  What do you mean they "have different routes"?  If it's an issue where some are ".delete" and some are ".destroy", then I think that's a different issue that should be changed as well.

**Comment by vblinden** [comment]
Yes, we generate all the routes for the datatypes. But we set the routes for database and setting our self. So they differ from the generated ones. I could make that a different pull request, where we make the routes the same?

**Comment by fletch3555** [comment]
Yes, please make them all the same (bread ones should be resource routes, so they're all `destroy`). We did this same thing a while back, so we must've just missed some.  Go ahead and test that under the 1.x-dev branch to make sure it wasn't just merged there for some reason

**Comment by fletch3555** [comment]
Yes, please make them all the same (bread ones should be resource routes, so they're all `destroy`). We did this same thing a while back, so we must've just missed some.  Go ahead and test that under the 1.x-dev branch to make sure it wasn't just merged there for some reason

**Comment by vblinden** [comment]
I made them all append (so we have some unity, like the resource routes). Are there any other delete modals that I missed?

**Review by fletch3555 — APPROVED** [review]

**Comment by emptynick** [comment]
Something went wrong [here](https://github.com/the-control-group/voyager/pull/2677/files#diff-52e64b52374bf2d87cd76ecbb043ca63R109)
You set the form-action to `database.destroy` when someone trys to delete a menu.
Good thing is that `table` is not defined, so this just doesn't work.
Otherwise it would just delete a whole table

**Comment by vblinden** [comment]
Wow! Good catch! Should I make a pr to fix this?

**Comment by emptynick** [comment]
No, I did already: https://github.com/the-control-group/voyager/pull/2892

**Pull request merged**